lib/xapi-stdext-std/listext.ml

```diff
@@ -196,4 +196,14 @@ module List = struct
         aux ((upper - 1) :: accu) (upper - 1)
     in
     aux []
+
+  let find_first_sorted compare =
+    let rec loop ref = function
```

Comment:
I think `ref` is a confusing name in OCaml because of references; 'pivot` could be an alternative.

Is this not the minimum of a list? Could this be implemented as a fold using `min`?

Reply:
yes a fold using a helper function that implements min using the given 'compare' would be shorter probably

Reply:
I suggest to use something with min or minimum as the name.

Reply:
Indeed, a fold works, `min` has to be defined in function of the compare function, to allow, for example, comparing strings.

I'll change its name to `find_minimum`, should make it less confusing

Reply:
How about just `minimum` - it's `List.minimum`; this fails for empty lists so there needs to be some exception.

Reply:
I like using the standard library as guidance, it has favoured options to exceptions in recent years, and it tends to use `find_` to extract single values (or [searching](https://v2.ocaml.org/api/List.html#1_Listsearching)) from lists.